### ISSUE TITLE
Allow to split `database.connection_string`

### ISSUE THREAD

**Issue by politician** [treeverse/charts#95]
## TLDR;

I need to be able to set `database.name`, `database.host`, `database.port`, `database.username`, `database.password` instead of using `database.connection_string`.

## Context

- I use GitOps with Flux
- I am deploying lakeFS using Helm on k8s
- I use [postgres-operator](https://github.com/zalando/postgres-operator) to create a db for lakeFS. 

postgres-operator nicely generates username/password that are stored in a k8s secret.

## Problem

AFAIK I cannot concatenate the values from a secret into a string dynamically, so I cannot generate a value for `database.connection_string`

### Helm only

For my particular use case, I can use [Flux's value override](https://fluxcd.io/docs/components/helm/helmreleases/#values-overrides) to use a value from a secret to inject in the `values.yaml`.
This would require however that each value has its own YAML entry such as this:

```yaml
secrets:
  database:
    name: lakefs
    host: localhost
    port: 5432
    username: postgres
    password: my-password
```

### Config + Env via Helm

A more generic approach would be to support this split also in the LakeFS [configuration file](https://docs.lakefs.io/reference/configuration.html#reference) and add support for setting environment variables in the chart so I could just do something like:

```yaml
env:
  - name: LAKEFS_DATABASE_PASSWORD
    valueFrom:
      secretKeyRef:
        name: some-secret
        key: password
```

What do you think?

**Comment by politician** [comment]
@johnnyaug this works great thank you!

Here's the relevant part of my `values.yaml` for posterity:

```yaml
secrets:
  databaseConnectionString: "postgres://lakefs-postgres:5432/lakefs"
extraEnvVars:
  - name: PGUSER
    valueFrom:
      secretKeyRef:
        name: lakefs.lakefs-postgres.credentials.postgresql.acid.zalan.do
        key: username
  - name: PGPASSWORD
    valueFrom:
      secretKeyRef:
        name: lakefs.lakefs-postgres.credentials.postgresql.acid.zalan.do
        key: password
```

`lakefs.lakefs-postgres.credentials.postgresql.acid.zalan.do` is my secret name and it contains two keys (`username` and `password`)
